components/raftstore/src/store/fsm/peer.rs

```diff
@@ -2061,6 +2061,14 @@ where
             return Ok(());
         }
 
+        if matches!(msg_type, MessageType::MsgUnreachable) {
```

Comment:
How can it receive the `MsgUnreachable`? Is there any test case to verify it's actually working?

Reply:
by checking whether the progress state to this peer is Probe, is ok?

Reply:
No, it's too hacky. The point of making it unreachable is to hint leader don't send any further messages. You should check the behavior instead of checking some properties.